internal/services/containers/kubernetes_cluster_resource_upgrade_test.go

```diff
@@ -279,29 +279,64 @@ func TestAccKubernetesCluster_upgradeSettings(t *testing.T) {
 
 	data.ResourceTest(t, r, []acceptance.TestStep{
 		{
-			Config: r.upgradeSettingsConfig(data, "2"),
+			Config: r.upgradeSettingsConfig(data, map[string]interface{}{
+				"max_surge":                "2",
+				"drain_timeout_in_minutes": 35,
+			}),
```

Comment:
These test steps could still be condensed further, I've added what the test steps should look like below:

```
data.ResourceTest(t, r, []acceptance.TestStep{
		{
			Config: r.upgradeSettingsConfig(data, map[string]interface{}{
				"max_surge":                    "2%",
				"drain_timeout_in_minutes":      35,
				"node_soak_duration_in_minutes": 18,
			}),
			Check: acceptance.ComposeTestCheckFunc(
				check.That(data.ResourceName).ExistsInAzure(r),
				check.That(data.ResourceName).Key("default_node_pool.0.upgrade_settings.#").HasValue("1"),
				check.That(data.ResourceName).Key("default_node_pool.0.upgrade_settings.0.max_surge").HasValue("2"),
				check.That(data.ResourceName).Key("default_node_pool.0.upgrade_settings.0.drain_timeout_in_minutes").HasValue("40"),
				check.That(data.ResourceName).Key("default_node_pool.0.upgrade_settings.0.node_soak_duration_in_minutes").HasValue("18"),
			),
		},
		data.ImportStep(),
		{
			Config: r.upgradeSettingsConfig(data, map[string]interface{}{
				"max_surge":                    "10%",
				"drain_timeout_in_minutes":      0,
				"node_soak_duration_in_minutes": 0,
			}),
			Check: acceptance.ComposeTestCheckFunc(
				check.That(data.ResourceName).ExistsInAzure(r),
				check.That(data.ResourceName).Key("default_node_pool.0.upgrade_settings.#").HasValue("1"),
				check.That(data.ResourceName).Key("default_node_pool.0.upgrade_settings.0.max_surge").HasValue("10%"),
				check.That(data.ResourceName).Key("default_node_pool.0.upgrade_settings.0.drain_timeout_in_minutes").HasValue("0"),
				check.That(data.ResourceName).Key("default_node_pool.0.upgrade_settings.0.node_soak_duration_in_minutes").HasValue("0"),
			),
		},
		data.ImportStep(),
		{
			Config: r.upgradeSettingsConfig(data, map[string]interface{}{
				"max_surge": "10%",
			}),
			Check: acceptance.ComposeTestCheckFunc(
				check.That(data.ResourceName).ExistsInAzure(r),
				check.That(data.ResourceName).Key("default_node_pool.0.upgrade_settings.#").HasValue("1"),
				check.That(data.ResourceName).Key("default_node_pool.0.upgrade_settings.0.max_surge").DoesNotExist(),
				check.That(data.ResourceName).Key("default_node_pool.0.upgrade_settings.0.drain_timeout_in_minutes").DoesNotExist(),
				check.That(data.ResourceName).Key("default_node_pool.0.upgrade_settings.0.node_soak_duration_in_minutes").DoesNotExist(),
			),
		},
		data.ImportStep(),
```

This will:
1. Test the create path for `drain_timeout_in_minutes` and `node_soak_duration_in_minutes`
2. Test the update path for `drain_timeout_in_minutes` and `node_soak_duration_in_minutes`
3. Test the update path for `drain_timeout_in_minutes` and `node_soak_duration_in_minutes` by removing it from the config

As a note, `max_surge` _needs_ to be specified at the moment since the API changed the default (this will be defaulted in the provider in 4.0). We get a plan diff if it isn't specified.